File: LeetCode Problems/201 to 300/228. Summary Ranges/228. Summary Ranges.java
class Solution {
    private List<String> ans;
    public List<String> summaryRanges(int[] nums) {
        ans = new ArrayList<>();
        if (nums.length == 0) return ans;
        long curr = nums[0], prev = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] - prev == 1) {
                prev = nums[i];
                continue;
            }
            insert(curr, prev);
            curr = prev = nums[i];
        }
        insert(curr, prev);
        return ans;
    }
    private void insert(long curr, long prev) {
        if (curr == prev)
            ans.add(String.valueOf(curr));
        else
            ans.add(String.valueOf(curr) + "->" + String.valueOf(prev));
    }
}

File: LeetCode Problems/201 to 300/228. Summary Ranges/228. Summary Ranges.cpp
typedef long long ll;

class Solution {
    vector<string> ans;

public:
    void insert(ll curr, ll prev) {
        if (curr == prev)
            ans.push_back(to_string(curr));
        else
            ans.push_back(to_string(curr) + "->" + to_string(prev));
    }
    vector<string> summaryRanges(vector<int> &nums) {
        if (nums.size() == 0) return {};
        ll curr = nums[0], prev = nums[0];
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] - prev == 1) {
                prev = nums[i];
                continue;
            }
            insert(curr, prev);
            curr = prev = nums[i];
        }
        insert(curr, prev);
        return ans;
    }
};